src/struct/hash.h

```diff
@@ -15,16 +15,17 @@
  */
 
 #include <stdint.h>
+#include <asm/bitsperlong.h>
 
 /* 2^31 + 2^29 - 2^25 + 2^22 - 2^19 - 2^16 + 1 */
 #define GOLDEN_RATIO_PRIME_32 0x9e370001UL
 /*  2^63 + 2^61 - 2^57 + 2^54 - 2^51 - 2^18 + 1 */
 #define GOLDEN_RATIO_PRIME_64 0x9e37fffffffc0001UL
 
-#if __WORDSIZE == 32
+#if __BITS_PER_LONG == 32
```

Comment:
I'd suggest `__LP64__` and `__ILP32__` per https://sourceforge.net/p/predef/wiki/DataModels/ .